### ISSUE TITLE
includePath using project.basedir is failing under certain conditions (Trac #586)

### ISSUE THREAD

**Issue by phing-issues-importer** [phingofficial/phing#537]
Summary:

If running `phing -f path/to/build.xml myTarget` and using an `<includePath>` that uses ${project.basedir}, the wrong path is added.

Details

As I understand it, project.basedir is set to the basedir of the build.xml file. But under some circumstances it seems to be set to the current working directory for part of the build, and to the basedir of the build.xml for the rest of the build.

In particular, when using <includepath classpath="${project.basedir}/foo"/> and then running Phing this way:

$ phing -f ../../build.xml foo

This seems to set `${project.basedir}` to `.` during the includpath execution, but then treat substitute the basedir of `${buildxml}` for the rest of the processing.

This seems to me to be a bug... or at least a nuance that should be documented.

Platform:

```
* phing 2.4.1
* PHP 5.3.3 
* Mac OS 10.6
```

Migrated from https://www.phing.info/trac/ticket/586

``` json
{
    "status": "new", 
    "changetime": "2016-10-07T08:28:55", 
    "description": "Summary:\n\nIf running `phing -f path/to/build.xml myTarget` and using an `<includePath>` that uses ${project.basedir}, the wrong path is added.\n\nDetails\n\nAs I understand it, project.basedir is set to the basedir of the build.xml file. But under some circumstances it seems to be set to the current working directory for part of the build, and to the basedir of the build.xml for the rest of the build.\n\nIn particular, when using <includepath classpath=\"${project.basedir}/foo\"/> and then running Phing this way:\n\n$ phing -f ../../build.xml foo\n\nThis seems to set `${project.basedir}` to `.` during the includpath execution, but then treat substitute the basedir of `${buildxml}` for the rest of the processing.\n\nThis seems to me to be a bug... or at least a nuance that should be documented.\n\nPlatform:\n\n    * phing 2.4.1\n    * PHP 5.3.3 \n    * Mac OS 10.6", 
    "reporter": "matt@aleph-null.tv", 
    "cc": "", 
    "resolution": "", 
    "_ts": "1475828935569836", 
    "component": "phing-tasks-system", 
    "summary": "includePath using project.basedir is failing under certain conditions", 
    "priority": "major", 
    "keywords": "", 
    "version": "2.4.1", 
    "time": "2010-11-12T22:41:08", 
    "milestone": "4.0", 
    "owner": "mrook", 
    "type": "defect"
}
```

**Comment by phing-issues-importer** [comment]
Trac update at 20101112T23:06:02: matt@aleph-null.tv commented:

> I made a mistake in my build.xml file that seems to have caused it.
> 
> With a little more examination, I noticed that `basedir="."` was set in the `<project>` element. And as expected, it overrode the project.basedir propery's default value.
> 
> Removing that solved the problem.
> 
> What I can't figure out, though, is why some of the paths resolved to the CWD for where I ran phing, and others to the basedir of build.xml. Again, though, that may be my fault. I have been using phing on dozens of projects with great success. And I think I've written about a dozen custom tasks. They're simple to create, and prove very useful.
> 
> The spam blocker is blocking me from submitting this, so I am adding a few lines of text to get it under the threshold for marking this as an invalid comment. You can safely ignore this text.

**Comment by phing-issues-importer** [comment]
Trac update at 20110220T16:43:36: mrook changed milestone from "2.4.5" to "TBD"

**Comment by phing-issues-importer** [comment]
Trac update at 20120130T22:11:46: mrook changed milestone from "TBD" to "2.5.0"

**Comment by phing-issues-importer** [comment]
Trac update at 20161007T08:28:55: 
* mrook commented:

> Milestone renamed


* mrook changed milestone from "3.0" to "4.0"

**Comment by mrook** [comment]
The confusion about `project.basedir` was fixed in #668.
